src/libpriv/rpmostree-origin.c

```diff
@@ -54,30 +58,40 @@ rpmostree_origin_parse_keyfile (GKeyFile         *origin,
   ret->refcount = 1;
   ret->kf = keyfile_dup (origin);
 
+  ret->cached_initramfs_args = g_ptr_array_new_with_free_func (g_free);
+
   /* NOTE hack here - see https://github.com/ostreedev/ostree/pull/343 */
   g_key_file_remove_key (ret->kf, "origin", "unlocked", NULL);
 
-  ret->unconfigured_state = g_key_file_get_string (ret->kf, "origin", "unconfigured-state", NULL);
+  ret->cached_unconfigured_state = g_key_file_get_string (ret->kf, "origin", "unconfigured-state", NULL);
 
-  ret->refspec = g_key_file_get_string (ret->kf, "origin", "refspec", NULL);
-  if (!ret->refspec)
+  ret->cached_refspec = g_key_file_get_string (ret->kf, "origin", "refspec", NULL);
+  if (!ret->cached_refspec)
     {
-      ret->refspec = g_key_file_get_string (ret->kf, "origin", "baserefspec", NULL);
-      if (!ret->refspec)
+      ret->cached_refspec = g_key_file_get_string (ret->kf, "origin", "baserefspec", NULL);
+      if (!ret->cached_refspec)
         {
           g_set_error (error, G_IO_ERROR, G_IO_ERROR_FAILED,
                        "No origin/refspec or origin/baserefspec in current deployment origin; cannot handle via rpm-ostree");
           return NULL;
         }
 
-      ret->packages = g_key_file_get_string_list (ret->kf, "packages", "requested", NULL, NULL);
+      ret->cached_packages = g_key_file_get_string_list (ret->kf, "packages", "requested", NULL, NULL);
     }
 
   /* Canonicalize NULL to an empty list for sanity */
-  if (!ret->packages)
-    ret->packages = g_new0 (char*, 1);
+  if (!ret->cached_packages)
+    ret->cached_packages = g_new0 (char*, 1);
 
-  ret->override_commit = g_key_file_get_string (ret->kf, "origin", "override-commit", NULL);
+  ret->cached_override_commit = g_key_file_get_string (ret->kf, "origin", "override-commit", NULL);
+
+  {
+    g_auto(GStrv) initramfs_args =
+      g_key_file_get_string_list (ret->kf, "rpmostree", "initramfs-args", NULL, NULL);
+    for (char **it = initramfs_args; it && *it; it++)
+      g_ptr_array_add (ret->cached_initramfs_args, g_strdup (*it));
+    g_ptr_array_add (ret->cached_initramfs_args, NULL);
```

Comment:
I think I had it this way because `GPtrArray` is more generic but still compatible. But really, it's premature at this point, until we add e.g. `rpmostree_origin_add_initramfs_arg()`.